Code/Editor/Core/EditorActionsHandler.cpp

```diff
@@ -0,0 +1,512 @@
+    }
+
+    {
+        AzToolsFramework::ToolBarProperties toolBarProperties;
+        toolBarProperties.m_name = "Play Controls";
+        m_toolBarManagerInterface->RegisterToolBar(PlayControlsToolBarIdentifier, toolBarProperties);
+    }
+
+    // Set the toolbars
+    m_mainWindow->addToolBar(Qt::ToolBarArea::TopToolBarArea, m_toolBarManagerInterface->GetToolBar(ToolsToolBarIdentifier));
+    m_mainWindow->addToolBar(Qt::ToolBarArea::TopToolBarArea, m_toolBarManagerInterface->GetToolBar(PlayControlsToolBarIdentifier));
+}
+
+QWidget* EditorActionsHandler::CreateDocsSearchWidget()
+{
+    QWidget* containerWidget = new QWidget(m_mainWindow);
+    auto lineEdit = new AzQtComponents::SearchLineEdit(m_mainWindow);
+    QHBoxLayout* layout = new QHBoxLayout;
+    layout->setContentsMargins(0, 0, 0, 0);
+    layout->addWidget(lineEdit);
+    containerWidget->setLayout(layout);
+    containerWidget->setContentsMargins(2, 0, 2, 0);
+    lineEdit->setPlaceholderText(QObject::tr("Search documentation..."));
+
+    auto searchAction = [lineEdit]()
+    {
+        auto text = lineEdit->text();
+        if (text.isEmpty())
+        {
+            QDesktopServices::openUrl(QUrl("https://www.o3de.org/docs/"));
+        }
+        else
+        {
+            QUrl docSearchUrl("https://www.o3de.org/search/");
+            QUrlQuery docSearchQuery;
+            docSearchQuery.addQueryItem("query", text);
+            docSearchUrl.setQuery(docSearchQuery);
+            QDesktopServices::openUrl(docSearchUrl);
+        }
+        lineEdit->clear();
+    };
+    QObject::connect(lineEdit, &QLineEdit::returnPressed, m_mainWindow, searchAction);
+
+    QMenu* helpMenu = m_menuManagerInterface->GetMenu(HelpMenuIdentifier);
+
+    QObject::connect(helpMenu, &QMenu::aboutToHide, lineEdit, &QLineEdit::clear);
+    QObject::connect(helpMenu, &QMenu::aboutToShow, lineEdit, &QLineEdit::clearFocus);
+
+    return containerWidget;
+}
+
+void EditorActionsHandler::OnViewPaneOpened(const char* viewPaneName)
+{
+    AZStd::string toolActionIdentifier = AZStd::string::format("o3de.action.tool.%s", viewPaneName);
+    m_actionManagerInterface->UpdateAction(toolActionIdentifier);
+}
+
+void EditorActionsHandler::OnViewPaneClosed(const char* viewPaneName)
+{
+    AZStd::string toolActionIdentifier = AZStd::string::format("o3de.action.tool.%s", viewPaneName);
+    m_actionManagerInterface->UpdateAction(toolActionIdentifier);
+}
+
+void EditorActionsHandler::RefreshToolActions()
+{
+    // If the tools are being displayed in the menu or toolbar already, remove them.
+    m_menuManagerInterface->RemoveActionsFromMenu(ToolsMenuIdentifier, m_toolActionIdentifiers);
+    m_toolBarManagerInterface->RemoveActionsFromToolBar(ToolsToolBarIdentifier, m_toolActionIdentifiers);
+    m_toolActionIdentifiers.clear();
+
+    AZStd::vector<AZStd::pair<AZStd::string, int>> toolsMenuItems;
+    AZStd::vector<AZStd::pair<AZStd::string, int>> toolsToolBarItems;
+    int sortKey = 10;
```

Comment:
nit: can we have a comment on why this constant is 10?

Reply:
Making a small change to this code, adding a comment to go with it :D